tests/test_compile.py

```diff
@@ -70,25 +72,73 @@ def test_compile_not_matching_targets(self):
 
     def test_compile_vars_target_missing(self):
         inventory_path = "inventory"
-        target_filename = "minikube-es"
-        target_obj = inventory_reclass(inventory_path)['nodes'][target_filename]['parameters']['kapitan']
-        # delete vars.target
-        del target_obj["vars"]["target"]
+        target_filename = "not-a-real-target"
+        inv = inventory_reclass(inventory_path)
 
         with self.assertRaises(InventoryError) as ie:
-            validate_matching_target_name(target_filename, target_obj, inventory_path)
+            validate_matching_target_name(target_filename, inv, inventory_path)
 
         error_message = "Target missing: target \"{}\" is missing parameters.kapitan.vars.target\n" \
                         "This parameter should be set to the target name"
         self.assertTrue(error_message.format(target_filename), ie.exception.args[0])
 
+    def test_inventory_reveal_or_compile(self):
+        inventory_path = "inventory"
+        refs_path = "refs"
+
+        # Build inventory
+        ref_controller = RefController("refs")
+        cached.ref_controller_obj = ref_controller
+        cached.revealer_obj = Revealer(ref_controller, reveal=False)
+
+        inv = inventory_reclass(inventory_path)
+        self.assertEqual(
+            inv["nodes"]["minikube-mysql"]["parameters"]["mysql"]["users"]["root"]["password"],
+            "?{gpg:targets/minikube-mysql/mysql/password:ec3d54de}",
+        )
+
+    def test_inventory_reveal_or_compile_with_targets(self):
+        inventory_path = "inventory"
+        refs_path = "refs"
+        target_filename = "minikube-es"
+
+        # Build inventory
+        ref_controller = RefController(refs_path)
+        cached.ref_controller_obj = ref_controller
+        cached.revealer_obj = Revealer(ref_controller, reveal=False, targets=[target_filename])
+        inv = inventory_reclass(inventory_path)
+
+        self.assertEqual(
+            inv["nodes"]["minikube-mysql"]["parameters"]["mysql"]["users"]["root"]["password"],
+            "?{gpg:targets/minikube-mysql/mysql/password|randomstr|base64}",
+        )
+        self.assertEqual(
+            inv["nodes"][target_filename]["parameters"]["elasticsearch"]["replicas"],
+            2,
+        )
+
+    def test_inventory_reveal_or_compile_with_invalid_ref(self):
+        inventory_path = "inventory"
+        refs_path = "refs"
+
+        # Build inventory
+        ref_controller = RefController(refs_path)
+        cached.ref_controller_obj = ref_controller
+        cached.revealer_obj = Revealer(ref_controller, reveal=False)
+        inv = inventory_reclass(inventory_path)
+        inv["nodes"]["minikube-es"]["secret"] = "?{gkms:secret/value}"
```

Comment:
I have the impression this is always going to fail anyway since gkms isn't tested on Travis yet. 
Maybe gpg would be a saner option here.

Reply:
It shouldn't matter since this test is for when a ref that is defined in the inventory cannot be found, i.e., it doesn't exist, there was a typo, etc.